src/util-lua-bytevarlib.c

```diff
@@ -0,0 +1,114 @@
+/* Copyright (C) 2025 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+#include "suricata-common.h"
+#include "detect-byte.h"
+#include "util-lua-common.h"
+#include "util-lua-bytevarlib.h"
+#include "util-lua.h"
+#include "detect-lua.h"
+#include "detect-lua-extensions.h"
+
+#include "lauxlib.h"
+
+static const char suricata_bytevar_mt[] = "suricata:bytevar:mt";
+
+static DetectLuaData *GetLuaData(lua_State *luastate)
+{
+    DetectLuaData *ld;
+    lua_pushlightuserdata(luastate, (void *)&luaext_key_ld);
+    lua_gettable(luastate, LUA_REGISTRYINDEX);
+    ld = lua_touserdata(luastate, -1);
+    return ld;
+}
+
+static int LuaBytevarEnsure(lua_State *L)
+{
+    const Signature *s = lua_touserdata(L, -2);
+    const char *name = luaL_checkstring(L, -1);
+    DetectLuaData *ld = GetLuaData(L);
+    if (ld->bytevars == DETECT_LUA_MAX_BYTEVARS) {
+        luaL_error(L, "too many bytevars registered");
```

Comment:
registered is "ensured" here? I do guess the name is a bit confusing.

Would `map` make sense? We're mapping a name to the id already set up before the script?

Reply:
Yeah, it would. Thats just the same error from previous git master, even though it was just creating a mapping as well.  'too many bytevars mapped`?